### ISSUE TITLE
I can't run main.py, No module named 'keras.src.engine'

### ISSUE THREAD

**Issue by 8maradona8** [kochlisGit/ProphitBet-Soccer-Bets-Predictor#77]
Hello, I am using macOS Sonoma M1chip, my python version is 3.11.4 ia m installing all modules from requirements navigate to folder but can't run it some problem with kerbs

P.S When try to install this version 
tensorflow~=2.9.1
tensorflow-addons~=0.17.1

I get this error 
ERROR: Could not find a version that satisfies the requirement tensorflow-addons==0.17.1 (from versions: 0.21.0, 0.22.0, 0.23.0)
ERROR: No matching distribution found for tensorflow-addons==0.17.1

ERROR: Could not find a version that satisfies the requirement tensorflow==2.9.1 (from versions: 2.13.0rc0, 2.13.0rc1, 2.13.0rc2, 2.13.0, 2.13.1, 2.14.0rc0, 2.14.0rc1, 2.14.0, 2.14.1, 2.15.0rc0, 2.15.0rc1, 2.15.0, 2.15.1, 2.16.0rc0, 2.16.1)
ERROR: No matching distribution found for tensorflow==2.9.1

then I install latest version on both modules
Please help me if you can

Thank You!

**Comment by looukass** [comment]
https://yashguptatech.medium.com/tensorflow-setup-on-apple-silicon-mac-m1-m1-pro-m1-max-661d4a6fbb77

**Comment by kochlisGit** [comment]
It seems tensorflow 2.9.1 is not compatible with python 3.11. It is in python 3.9 though. What You can do is actually install a different tensoflow version, e.g. 2.9.0 or 2.10 maybe

**Comment by kochlisGit** [comment]
ok. then download python 3.9. It should work!

**Comment by Fx5flier** [comment]
Got 3.9 and set it up , this is what I've got:
ERROR: Could not find a version that satisfies the requirement tensorflow==2.9.1 (from versions: 2.13.0rc0, 2.13.0rc1, 2.13.0rc2, 2.13.0, 2.13.1, 2.14.0rc0, 2.14.0rc1, 2.14.0, 2.14.1, 2.15.0rc0, 2.15.0rc1, 2.15.0, 2.15.1, 2.16.0rc0, 2.16.1)
ERROR: No matching distribution found for tensorflow==2.9.1

**Comment by 8maradona8** [comment]
For me, situation is the same

**Comment by kochlisGit** [comment]
What is your operating system? Are you using windows?
Also, what about tensorflow 2.13.0? Does it work?

**Comment by 8maradona8** [comment]
Like i say in first post, i am with macos and the error is with 2.16 tensorflow, after that i downgrade python to 3.9.6 and try to reinstall with tensorflow 2.9.1 but same error like this:
ERROR: Could not find a version that satisfies the requirement tensorflow==2.9.1 (from versions: 2.13.0rc0, 2.13.0rc1, 2.13.0rc2, 2.13.0, 2.13.1, 2.14.0rc0, 2.14.0rc1, 2.14.0, 2.14.1, 2.15.0rc0, 2.15.0rc1, 2.15.0, 2.15.1, 2.16.0rc0, 2.16.1)
ERROR: No matching distribution found for tensorflow==2.9.1

**Comment by mnomoko** [comment]
I faced the same issue !
On Mac OS you have python installed for Xcode settings. The installed version was python 3.9, but for a mysterious reason I had an error.

I used Homebrew to install and test differents versions of python but nothing changed.
So I installed "pyenv" (still from Homebrew) and manage my python's versions within instead of using Homebrew directly to install or link python.
To resolve my problem I installed python v3.9.18 and that's all. I do not modifIed nothing else.

My hypothesis is that the version of python from Xcode is a minimal version or a version that is too old

**Comment by rodrigom03** [comment]
> Like i say in first post, i am with macos and the error is with 2.16 tensorflow, after that i downgrade python to 3.9.6 and try to reinstall with tensorflow 2.9.1 but same error like this: ERROR: Could not find a version that satisfies the requirement tensorflow==2.9.1 (from versions: 2.13.0rc0, 2.13.0rc1, 2.13.0rc2, 2.13.0, 2.13.1, 2.14.0rc0, 2.14.0rc1, 2.14.0, 2.14.1, 2.15.0rc0, 2.15.0rc1, 2.15.0, 2.15.1, 2.16.0rc0, 2.16.1) ERROR: No matching distribution found for tensorflow==2.9.1

Same error here. macOS M2 and I can't figure out a solution. installed pyenv and made the v3.9.18 global and still does not work. You got the solution?

**Comment by 8maradona8** [comment]
No, still cant run it, try to reinstall everything fresh, but same error :(

**Comment by Calin96** [comment]
Fix that I’ve found that it works is:
 Change to a non-mac laptop. I have this already running on my other win laptop.

**Comment by rodrigom03** [comment]
Yes you’re right. Yesterday after endlessly trying to make it work on Mac, just switched to a Windows laptop and worked perfectly.

**Comment by rodrigom03** [comment]
Got the solution for Mac users! Use this command:

pip install tensorflow-macos==2.9.1

**Comment by KnightRider37** [comment]
> Got the solution for Mac users! Use this command:
> 
> pip install tensorflow-macos==2.9.1

worked like a charm! thanks!

**Comment by kkaral2** [comment]
Successfully created a conda environment on Apple M3 but when launching the main.py I am getting an empty screen:
<img width="1197" alt="empty_screen" src="https://github.com/user-attachments/assets/03ffe970-a41f-45d3-91eb-0a719a96d1d2" />
Any advice?
